en/components/grid.md

```diff
@@ -22,345 +22,503 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
         </ng-template>
     </igx-column>
-</igx-grid>
-```
+    ...
 
-Column properties can also be set in code in the **initColumns** event:
+In the snippet above we "take" a reference to the implicitly provided cell value. This is sufficient if you just want to present some data and maybe apply some custom styling or pipe transforms over the value of the cell. However even more useful is to take the `IgxGridCellComponent` object itself as shown below:
+
+    <igx-grid #grid [data]="data">
+        <igx-column dataType="string" field="Name">
+            <ng-template igxCell let-cell="cell">
+                <!-- Implement row deleting inside the cell template itself -->
+                <span tabindex="0" (keydown.delete)="grid.deleteRow(cell.rowIndex)">{{ cell.value | titlecase }}</span>
+            </ng-template>
+        </igx-column>
+        <igx-column dataType="boolean" field="Subscribtion">
+            <ng-template igxCell let-cell="cell">
+                <!-- Bind the cell value through the ngModel directive and update the data source when the value is changed in the template -->
+                <input type="checkbox" [ngModel]="cell.value" (ngModelChange)="cell.update($event)" />
+            </ng-template>
+        </igx-column>
+    <igx-grid>
+
+The column also accepts one last template that will be used when a cell is in edit mode. As with the other column templates, the provided context object is again the cell value and the cell object itself. Of course in order to have your template show up make sure to set the `editable` property of the column to true.
+
+    <igx-column dataType="number" editable="true" field="Price">
+        <ng-template igxCellEditor let-cell="cell">
+            <label for="price">
+                Enter the new price tag
+            </label>
+            <input name="price" type="number" [ngModel]="cell.value" (ngModelChange)="cell.update(convertToNumber($event))" />
+        </ng-template>
+    </igx-column>
 
-```typescript
-public initColumns(event: IgxGridColumnInitEvent) {
-    const column: IgxColumnComponent = event.column;
-    if (column.field === 'ProductName') {
-      column.filtering = true;
-      column.sortable = true;
-      column.editable = true;
-    }
-}
-```
+Make sure to check the API for the [IgxGridCellComponent](#igxgridcellcomponent) in order to get accustomed with the provided properties you can use in your templates.
 
-The code above will make the **ProductName** column sortable, filterable, and editable and will instantiate the corresponding features UI (like inputs for editing, save dialogs, etc.).
+Each of the column templates can be changed programmatically at any point through the column object itself. For example in the code below, we have declared two templates for our user data. In out Typescript code we'll get references to the templates itself and then based on some condition we will render the appropriate for the column in our application.
```

Comment:
In *our* Typescript code we'll get references to the template *themselves*
-or-
... to the template itself (if there is only one)

...we will render the appropriate [word is missing here] for the column.  Render the appropriate template?